docker/test_env/erigon.go

```diff
@@ -0,0 +1,257 @@
+package test_env
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"html/template"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+
+	tc "github.com/testcontainers/testcontainers-go"
+	tcwait "github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/blockchain"
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+	"github.com/smartcontractkit/chainlink-testing-framework/logging"
+	"github.com/smartcontractkit/chainlink-testing-framework/utils/testcontext"
+)
+
+const (
+	//TODO use Tate's mirror?
+	ERIGON_IMAGE_TAG = "v2.54.0"
+)
+
+type Erigon struct {
+	EnvComponent
+	ExternalHttpUrl      string
+	InternalHttpUrl      string
+	ExternalWsUrl        string
+	InternalWsUrl        string
+	InternalExecutionURL string
+	ExternalExecutionURL string
+	generatedDataHostDir string
+	chainConfg           *EthereumChainConfig
+	consensusLayer       ConsensusLayer
+	l                    zerolog.Logger
+	t                    *testing.T
+}
+
+func NewErigon(networks []string, chainConfg *EthereumChainConfig, generatedDataHostDir string, consensusLayer ConsensusLayer, opts ...EnvComponentOption) *Erigon {
+	g := &Erigon{
+		EnvComponent: EnvComponent{
+			ContainerName: fmt.Sprintf("%s-%s", "erigon", uuid.NewString()[0:8]),
+			Networks:      networks,
+		},
+		chainConfg:           chainConfg,
+		generatedDataHostDir: generatedDataHostDir,
+		consensusLayer:       consensusLayer,
+		l:                    log.Logger,
+	}
+	for _, opt := range opts {
+		opt(&g.EnvComponent)
+	}
+	return g
+}
+
+func (g *Erigon) WithTestInstance(t *testing.T) *Erigon {
```

Comment:
Is there a reason we are not keeping the same naming consistency with existing containers? We have not been using WithTestInstance, we have been using WithTestLogger. We should keep consistent across the board for this kind of stuff, if we want to go with WithTestInstance (which I agree is a better name) then we need to either update all the others now or create a ticket to do it soon. We don't want to live with two patterns for too long.

Reply:
I changed it only, because it was a misleading name for me, because it was called `WithTestLogger` but we weren't passing logger, but `t`.

If you're okay with it, I'll change all the "old" methods also to the new one in this PR.

Reply:
@tateexon I renamed everything to `WithTestInstance`

Reply:
+1 for `WithTestInstance` name as testing.T is passed there. But, @Tofel do we really need whole `*testing.T`. If this is only for logging maybe you can initialize the logger with `testing.T`?